src/query-builder/SelectQueryBuilder.ts

```diff
@@ -4236,9 +4237,40 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
                     .join(" OR ")
             }
         } else {
-            const andConditions: string[] = []
-            for (const key in where) {
-                if (where[key] === undefined || where[key] === null) continue
+            let andConditions: string[] = []
+            for (let key in where) {
+                const parameterValue = where[key]
+
+                if (parameterValue === undefined) {
+                    if (this.expressionMap.throwOnUndefinedInFind) {
+                        throw new TypeORMError(
+                            `Undefined value encountered in property '${alias}.${key}' of the find operation. ` +
+                                `Set 'throwOnUndefinedInFind' to false in connection options to skip properties with undefined values.`,
+                        )
+                    }
+                    continue
+                }
+
+                if (parameterValue === null) {
+                    if (!this.expressionMap.treatJsNullAsSqlNull) {
+                        continue // Skip null values by default
+                    }
+
+                    // If treatJsNullAsSqlNull is true, transform JS null to SQL NULL
+                    const propertyPath = embedPrefix
+                        ? embedPrefix + "." + key
+                        : key
+                    const column =
+                        metadata.findColumnWithPropertyPathStrict(propertyPath)
+                    if (column) {
+                        let aliasPath = `${alias}.${propertyPath}`
+                        if (column.isVirtualProperty && column.query) {
+                            aliasPath = `(${column.query(alias)})`
+                        }
+                        andConditions.push(`${aliasPath} IS NULL`)
+                        continue
+                    }
```

Comment:
Could this code be moved into the existing `if (column)` block at line 4289? I think this would be more readable and avoid some duplication.

Reply:
Yeah, for sure, great catch!